vendor/vendor-ios/src/main/kotlin/com/malinskiy/marathon/ios/DockerTestParser.kt

```diff
@@ -0,0 +1,49 @@
+package com.malinskiy.marathon.ios
+
+import com.malinskiy.marathon.log.MarathonLogging
+import java.io.BufferedReader
+import java.io.File
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.util.concurrent.Executors
+
+interface DockerTestParser {
+    fun listTests(testRunnerPaths: List<File>): List<String>
+}
+
+class BinaryTestParser(private val binaryParserDockerImageName: String): DockerTestParser {
+    private val logger = MarathonLogging.logger(BinaryTestParser::class.java.simpleName)
+
+    private class DockerOutputReader(private val inputStream: InputStream,
+                                     private val consumer: (String) -> Unit) : Runnable {
+        override fun run() {
+            BufferedReader(InputStreamReader(inputStream)).lines()
+                    .forEach(consumer)
+        }
+    }
+
+    override fun listTests(testRunnerPaths: List<File>): List<String> {
+        val pathMappings = testRunnerPaths.map {
+            it.absolutePath to File("/tmp").resolve(it.name).absolutePath }
+
+        val command =
+                listOf("docker", "run", "--rm") +
```

Comment:
Later we can think about switching to https://github.com/docker-java/docker-java